references/common/metrics.go

```diff
@@ -127,47 +227,57 @@ func ListApplicationPods(c client.Client, app *appv1beta1.Application, component
 	return pods
 }
 
-// LoadApplicationMetrics load application metrics
-func (appMetrics *ApplicationMetricsStatus) LoadApplicationMetrics(c client.Client, conf *rest.Config, pods []v1.Pod) {
-	metricsClient := metricsclientset.NewForConfigOrDie(conf)
-	cpu, memory, stroage := uint64(0), uint64(0), uint64(0)
-
+// GetApplicationMetricsStatus get application metrics status
+func GetApplicationMetricsStatus(c client.Client, conf *rest.Config, pods []v1.Pod) *ApplicationMetricsStatus {
+	metricsStatus := &ApplicationMetricsStatus{}
+	cpuUsage, memoryUsage, storage := int64(0), int64(0), int64(0)
+	cpuLimit, cpuRequest := int64(0), int64(0)
+	memLimit, memRequest := int64(0), int64(0)
 	for _, pod := range pods {
-		podMetrics, err := GetPodMetrics(metricsClient, pod, false)
+		podMetrics, err := GetPodMetrics(conf, pod.Name, pod.Namespace, "")
```

Comment:
why is `cluster` empty? `GetPodMetrics` need cluster parameter to retrieve metrics

Reply:
Sorry, I don't have some good ideas to solve this problem. Could you give me some advice about it? I will be very grateful.

If I want to get pods in all clusters, is there any better method instead of trying to call the `GetPodMetrics` function with this `cluster` parameter with each cluster of an application.

Reply:
I want to get a better writing method to solve it.

Reply:
i have fixed it.